lib/Extension/Laravel/Adapter/Laravel/LaravelContainerInspector.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Phpactor\Extension\Laravel\Adapter\Laravel;
+
+use Phpactor\WorseReflection\Core\TypeFactory;
+use Phpactor\WorseReflection\Core\Type\ClassType;
+
+class LaravelContainerInspector
+{
+    private array $services = [];
+    public function __construct(private string $executablePath, private string $projectRoot)
+    {
+    }
+
+    public function service(string $id): ?ClassType
+    {
+        foreach ($this->services() as $short => $service) {
+            if ($short === $id || $service === $id) {
+                return TypeFactory::fromString('\\' . $service);
+            }
+        }
+        return null;
+    }
+
+
+    public function services(): array
+    {
+        if ([] === $this->services) {
+            $output = shell_exec("php $this->executablePath container $this->projectRoot");
```

Comment:
preferred to use the Symfony `Process` class

Reply:
also wonder if there is a way to inspect the container that doesn't involve an external binary?

Reply:
or if something like [this](https://github.com/barryvdh/laravel-ide-helper#phpstorm-meta-for-container-instances) is more idiomatic? do most devs use those generators?

Reply:
I intentionally did not go for that direction as it enforces a project dependency.

My package does use that generator and can generate it in the vendor folder.

With a lot of code I think we can figure it out. However, I personally would like to not focus on that for now.

Reply:
Switched to Process